src/LiquidHaskell/Cabal.hs

```diff
@@ -184,6 +199,35 @@ getUserArgs desc bi =
         Left err   -> dieNoVerbosity $
           "failed to parse LiquidHaskell options for " ++ desc ++ ": " ++ err
 
+--------------------------------------------------------------------------------
+-- Filter out files for which to run LiquidHaskell -----------------------------
+--------------------------------------------------------------------------------
+
+data CheckedFiles =
+    All
+  | Whitelist [FilePath]
+
+filterCheckedFiles :: CheckedFiles -> [FilePath] -> [FilePath]
+filterCheckedFiles All                 fps = fps
+filterCheckedFiles (Whitelist allowed) fps = intersectBy prefixOrEqual fps allowed
+  where
+    prefixOrEqual x y = x == y || y `isPrefixOf` x
+
+getCheckedFiles :: BuildInfo -> CheckedFiles
+getCheckedFiles bi =
+  maybe All (Whitelist . splitOn ' ') $ lookup liquidHaskellCheckedFiles (customFieldsBI bi)
+  where
+    splitOn :: Char -> String -> [String]
+    splitOn _ []  = []
+    splitOn c str = let (pref, suff) = break (== c) str
+                     in pref : splitOn c (drop 1 suff)
+
+verifyCheckedFiles :: CheckedFiles -> IO ()
+verifyCheckedFiles All = pure ()
+verifyCheckedFiles (Whitelist allowed) = for_ allowed $ \file -> do
+  exists <- doesPathExist file
+  unless exists $ dieNoVerbosity $ file ++ " specified in " ++ liquidHaskellCheckedFiles ++ " is missing!"
```

Comment:
This was initially a warning, but in trying to use it I found it *very* hard to see and made it a crash.

Reply:
:+1: